src/Mobs/Path.cpp

```diff
@@ -51,6 +51,14 @@ cPath::cPath(
 	m_Source.x = FloorC(a_StartingPoint.x - HalfWidthInt);
 	m_Source.y = FloorC(a_StartingPoint.y);
 	m_Source.z = FloorC(a_StartingPoint.z - HalfWidthInt);
+	
+	if ((m_Source.y <= 0) || (m_Destination.y <= 0)
+		|| (m_Source.y > 256) || (m_Destination.y > 256))
```

Comment:
That would add extra checks which are spread in cPath, imho it's much more error prone and less future-proof. Why not simply redefine what `IsSolid` does and avoid code dupes?

> that's making the function do something that it isn't supposed to do.

I like to look at it differently: That makes the function assume there's a solid roof just above the world and a solid ground just below the world.